### PR TITLE
UniStore: add FoldersCreate Endpoint test

### PR DESCRIPTION
**Pull request by filewalkwithme**
**What is this feature?**

This PR introduces the counterpart of [TestFoldersCreateAPIEndpoint](https://github.com/grafana/grafana/blob/b68b69c2b423d6057851f5839fc2d39c397dc069/pkg/api/folder_test.go#L34), but for the new endpoint backed by Unified Storage

**Why do we need this feature?**

We want to ensure that the CreateFolder endpoint backed by Unified Storage behaves in the same was as the Legacy Endpoint

**Who is this feature for?**

@grafana/grafana-search-and-storage 

**Which issue(s) does this PR fix?**:

Part of #https://github.com/grafana/search-and-storage-team/issues/110

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/contribute/release-notes/#how-to-determine-if-content-belongs-in-whats-new), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/contribute/release-notes/) doc.

### CODE REVIEW

File: pkg/tests/apis/folder/folders_test.go [hunk 1]
```diff
@@ -639,3 +644,167 @@ func TestIntegrationFolderCreatePermissions(t *testing.T) {
...
+			Resource:          "folders",
+			ResourceAttribute: "uid",
+			ResourceID:        "*",
+		},
+	}
+
+	// NOTE: folder creation does not return ErrFolderAccessDenied neither ErrFolderNotFound
+	tcs := []testCase{
+		{
+			description:  "folder creation succeeds given the correct request for creating a folder",
```
Comment (leonorfmartins) [inline pkg/tests/apis/folder/folders_test.go:682]: Just to confirm, creating subfolders is not possible, correct?

Comment (filewalkwithme) [reply pkg/tests/apis/folder/folders_test.go:682]: Creating subfolders is currently working both in Legacy and Unified Storage.

Why?

Comment (leonorfmartins) [reply pkg/tests/apis/folder/folders_test.go:682]: would it be better to add an additional test case for it as well? Currently we are only testing the happy path for `folderWithoutParentInput`

Comment (filewalkwithme) [reply pkg/tests/apis/folder/folders_test.go:682]: The idea for this test is just to be a counterpart of [TestFoldersCreateAPIEndpoint](https://github.com/grafana/grafana/blob/b68b69c2b423d6057851f5839fc2d39c397dc069/pkg/api/folder_test.go#L34)

If we are going to add a testcase for the creation of nested folders, then we could do it in a follow-up PR. What do you think?

**Review by leonorfmartins — APPROVED** [review]

**Pull request merged**